web/channels/player_channel.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Yggdrasil.PlayerChannel do
+  use Yggdrasil.Web, :channel
+  alias Yggdrasil.Player
+  alias Yggdrasil.Endpoint
+
+  def join("player:" <> user_id = topic, _message, socket) do
+    if socket.assigns.user == String.to_integer(user_id) do
+      socket = assign socket, :player_topic, topic
+      {:ok, socket}
+    else
+      {:error, %{ error: "auth failure"} }
+    end
+  end
+
+  def handle_in("join_game", _message, socket) do
+    player_topic = socket.assigns.player_topic
+    push_msg = fn (payload) ->
+      Endpoint.broadcast! player_topic, "event", payload
+    end
+
+    case Player.join_game socket.assigns.user, self, push_msg do
```

Comment:
I like this being pushed to the channel, but do have a question why not do this on `PlayerChannel.Join`? The scenario I'm thinking of is that say you have two tabs open both join the topic, but one doesn't send the "join_game" event. Any notifications pushed to the the topic would be received by anyone who has joined that topic (i know it's authenticated by user_id of course so really only talking about the user joining twice).

For example this could probably be proved by logging in, then opening dev tools and manually joining the "player:1" topic if that was the id and add a function to log anything broad cast to that channel, I bet you would not only see the page show the notifications, but also in the spoofed one as well in the console.